### PR TITLE
*: Fix xaggregate with index scan panic

### PR DESCRIPTION
**Pull request by shenli**
@coocood @zimulala

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

File: session_test.go [hunk 1]
```diff
@@ -2268,3 +2268,16 @@ func (s *testSessionSuite) TestRetryAttempts(c *C) {
...
+func (s *testSessionSuite) TestXAggregateWithIndexScan(c *C) {
+	initSQL := `
+		drop table IF EXISTS t;
+		CREATE TABLE t(col3 INT, index c3 (col3));
+		INSERT INTO t VALUES(1), (null), (2);`
+	store := newStore(c, s.dbName)
+	se := newSession(c, store, s.dbName)
+	mustExecMultiSQL(c, se, initSQL)
+
+	sql := "SELECT DISTINCT + - COUNT( col3 ) AS col1 FROM t AS cor0 WHERE col3 IS NOT NULL;"
```
Comment (zimulala) [inline session_test.go]: Simplify sql?

Comment (shenli) [inline session_test.go]: PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**